runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/PipelineResourcesTest.java

```diff
@@ -44,24 +49,98 @@
   @Rule public transient TemporaryFolder tmpFolder = new TemporaryFolder();
   @Rule public transient ExpectedException thrown = ExpectedException.none();
 
+  ClassLoader currentContext = null;
```

Comment:
nit: Consider making this a test rule like [RestoreSystemProperties](https://github.com/apache/beam/blob/master/sdks/java/core/src/test/java/org/apache/beam/sdk/testing/RestoreSystemProperties.java)